client/helpers/queued-jobs.ts

```diff
@@ -14,84 +25,132 @@ export interface PollQueuedJobReq {
   readonly id: string;
 
   /** Function called to get queued job. */
-  readonly getQueuedJob: (id: string) => Promise<AxiosResponse<QueuedJob>>;
+  readonly getQueuedJob: (
+    id: string,
+    cancelToken: CancelToken
+  ) => Promise<AxiosResponse<QueuedJob>>;
 
   /** If `true`, doesn't fail if API returns 404 status code. */
   readonly allow404?: boolean;
 
+  readonly limit?: Limit;
+
   /** {@link Polling} */
   readonly polling: Polling;
 }
 
 export interface PollQueuedJobRes<T> extends PollJobRes<T> {
-  attempts: number;
-  id: string;
+  readonly attempts: number;
+  readonly id: string;
 }
 
 type PollRes<T> = Failure | QueuedJob | T;
 
 interface PollJobRes<T> {
-  res: PollRes<T>;
-  status: number;
+  readonly res: PollRes<T>;
+  readonly status: number;
 }
 
+const ConnectTimeoutMs = 8000;
+const ClientErrorId = 'node-client-error';
+const Debug = true;
+
 /**
  * Poll `getQueuedJob` until redirected to resulting resource, `error`, or reach
  * `polling.maxAttempts`.
  *
  * @param req - {@link PollQueuedJobReq}.
  * @returns {@link PollQueuedJobRes}.
  */
-export async function pollQueuedJob<T extends { data: { id: string } }>({
+export async function pollQueuedJob<T>({
   id,
   getQueuedJob,
   allow404 = false,
+  limit,
   polling: { intervalMs, maxAttempts },
 }: PollQueuedJobReq): Promise<PollQueuedJobRes<T>> {
-  function poll(ms: number): Promise<PollJobRes<T>> {
-    return new Promise((resolve) => {
-      setTimeout(
-        () =>
-          getQueuedJob(id)
-            .then((r) => resolve({ status: r.status, res: r.data }))
-            .catch((error) => {
-              console.log(
-                `pollQueuedJob error, continuing. '${error.message}'`
-              );
-              const errors = new Set<ApiError>();
-              errors.add({
-                status: '503',
-                code: 'ServiceUnavailable',
-                title: 'Node client caught error in pollQueuedJob.',
-                detail: error.message,
-              });
-              resolve({ status: 500, res: { errors } });
-            }),
-        ms
+  async function poll(attempt: number): Promise<PollJobRes<T>> {
+    const cancelSrc = axios.CancelToken.source();
+    const timerId = setTimeout(
+      () => cancelSrc.cancel(`Connect timeout after ${ConnectTimeoutMs}ms.`),
+      ConnectTimeoutMs
+    );
```

Comment:
Add an 8 second `ConnectTimeout`. If we haven't connected in that amount of time, all kinds of errors start getting thrown. So instead, catch it early, use the cancel token to cancel the `axios` request, and retry.